proposals/0046-i18n-domain-support.md

```diff
@@ -0,0 +1,87 @@
+- Start Date: 2023/12/21
+- Reference Issues:
+- Implementation PR: https://github.com/withastro/astro/pull/9143
+
+
+# Summary
+
+First-class support for domain support in i18n routing.
+
+
+# Background & Motivation
+
+Websites that support internationalisation have different requirements. Among these requirements, there's the need to 
+have localised content under different subdomains or domains. 
+
+# Goals
+
+- Support different domains, all driven by the same Astro project;
+- Configure some locales to use only path prefixes and some locales to use domains;
+- Mix locales that require a different domain, with locales that don't require a different domain;
+
+# Non-Goals
+
+- Redirect users from a path website to a domain website 
+- Change website/domain based on the language of the user's browser 
+- Support for static/hybrid output
+- Offer the means to configure multiple domains for serverless hosts (Vercel, Netlify, etc.)
+
+
+# Detailed Design
+
+There are some restrictions about how the feature will initially work:
+- the feature works only with `output: "server"`, hence only in SSR;
+- the presence of at least one pre-rendered route will cause a failure in the build;
+- the option `base` is required, because Astro needs to know how to create absolute URLs for locales that aren't mapped to a domain;
```

Comment:
Is this required? Couldn’t the default `base` of `'/'` be used for this purpose? (Also noting that the code examples below do not include `base`, so would presumably invalid if this is true.)

Reply:
Oh, shoot, the wrong option! The required options is `site`! :D